bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/handler/ElroConnectsBridgeHandler.java

```diff
@@ -0,0 +1,862 @@
+    private short msgId;
+
+    private NetworkAddressService networkAddressService;
+
+    // Used when restarting connection, delay restart for 1s to avoid high network traffic
+    private volatile boolean restart;
+    static final int RESTART_DELAY_MS = 1000;
+
+    private volatile String connectorId = "";
+    // Used for getting IP address and keep connection alive messages
+    private static final String QUERY_BASE_STRING = "IOT_KEY?";
+    private volatile String queryString = QUERY_BASE_STRING + connectorId;
+    // Regex to retrieve ctrlKey from response on IP address message
+    private static final Pattern CTRL_KEY_PATTERN = Pattern.compile("KEY:([0-9a-f]*)");
+
+    private int refreshInterval = 60;
+    private volatile @Nullable InetAddress addr;
+    private volatile String ctrlKey = "";
+
+    private volatile @Nullable DatagramSocket socket;
+    private volatile @Nullable DatagramPacket ackPacket;
+
+    private volatile @Nullable ScheduledFuture<?> syncFuture;
+    private volatile @Nullable CompletableFuture<Boolean> awaitResponse;
+
+    private ElroConnectsDynamicStateDescriptionProvider stateDescriptionProvider;
+
+    private final Map<Integer, String> scenes = new ConcurrentHashMap<>();
+    private final Map<Integer, ElroConnectsDevice> devices = new ConcurrentHashMap<>();
+    private final Map<Integer, ElroConnectsDeviceHandler> deviceHandlers = new ConcurrentHashMap<>();
+
+    private int currentScene;
+
+    // We only keep 2 gson adapters used to serialize and deserialize all messages sent and received
+    private final Gson gsonOut = new Gson();
+    private Gson gsonIn = new Gson();
+
+    public ElroConnectsBridgeHandler(Bridge bridge, NetworkAddressService networkAddressService,
+            ElroConnectsDynamicStateDescriptionProvider stateDescriptionProvider) {
+        super(bridge);
+        this.networkAddressService = networkAddressService;
+        this.stateDescriptionProvider = stateDescriptionProvider;
+
+        resetScenes();
+    }
+
+    @Override
+    public void initialize() {
+        ElroConnectsBridgeConfiguration config = getConfigAs(ElroConnectsBridgeConfiguration.class);
+        connectorId = config.connectorId;
+        refreshInterval = config.refreshInterval;
+
+        if (connectorId.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Device ID not set");
+            return;
+        } else if (!CONNECTOR_ID_PATTERN.matcher(connectorId).matches()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Device ID not of format ST_xxxxxxxxxxxx with xxxxxxxxxxxx the lowercase MAC address of the connector");
+            return;
+        }
+
+        this.addr = null;
+        String ipAddress = config.ipAddress;
+        if (!ipAddress.isEmpty()) {
+            try {
+                InetAddress addr = InetAddress.getByName(ipAddress);
```

Comment:
`InetAddress.getByName` is a network operation and can block `intialize()`, which should return fast. You could resolve it in `startCommunication()`.

Reply:
Updated